ksqldb-common/src/main/java/io/confluent/ksql/schema/ksql/LogicalSchema.java

```diff
@@ -263,6 +281,11 @@ private LogicalSchema rebuild(
     if (withPseudoAndKeyColsInValue) {
       builder.add(Column.of(ROWTIME_NAME, ROWTIME_TYPE, VALUE, valueIndex++));
 
+      if (pseudoColumnVersion > 0) {
+        builder.add(Column.of(ROWOFFSET_NAME, ROWOFFSET_TYPE, VALUE, valueIndex++));
```

Comment:
Understanding this code requires knowledge that pseudocolumn version 0 means rowtime is the only pseudocolumn, whereas pseudocolumn version 1 means rowpartition and rowoffset should be added too. This is "magic" knowledge that should be avoided.

How about we add new constants such as `ROWTIME_PSEUDOCOLUMN_VERSION = 0` and `ROWPARTITION_ROWOFFSET_PSEUDOCOLUMN_VERSION = 1` so that this logic can instead become
```
if (pseudoColumnVersion >= ROWTIME_PSEUDOCOLUMN_VERSION) {
  // add in rowtime
}
if (pseudoColumnVersion >= ROWPARTITION_ROWOFFSET_PSEUDOCOLUMN_VERSION) {
  // add in rowpartition and rowoffset
}
```

Reply:
This would conflict with your above suggestion to make the ROWPARTITION/ROWOFFSET constant 0 for now, and both conditionals would always trigger. I could make a note to do this when the feature ships?